### DISCUSSION TITLE
Multipoint constraint

### DISCUSSION THREAD

**Discussion by muhammedfurkanyilmaz** [KratosMultiphysics/Kratos#12169] [Q&A]
Hi  @KratosMultiphysics/geomechanics,
How can I define multipoint constraint for uPw solution?

**Comment by avdg81** [comment]
Hi @muhammedfurkanyilmaz,

Thank you for your question. We will look into this and provide you with an answer when we know how this can be done (if possible).

Kind regards,
Anne

**Reply by avdg81** [reply]
Hi @muhammedfurkanyilmaz,

We have never used the multipoint constraints in the GeoMechanicsApplication ourselves. There is, however, [some documentation](https://kratosmultiphysics.github.io/Kratos/pages/Kratos/For_Developers/Data_Structures/MultiPointConstraint_Interface.html) which explains how it can be used in Kratos (just in case you had not found this page yourself). Also, I have found a test in the StructuralMechanicsApplication that may help you to get started: see `applications/StructuralMechanicsApplication/tests/test_eigen_solver_with_constraints.py`. Perhaps this already provides you with sufficient information to continue your analyses. If not, we are willing to provide further assistance. We would, however, in that case like to receive some (simplified) input files to give us the opportunity to experiment with your model and see whether we get things to work the way you would like them to work. Thank you.

Kind regards,
Anne

**Reply by muhammedfurkanyilmaz** [reply]
Thanks, @avdg81.
Let me check them and try to make a model. Then I'll inform you about the situation.

**Comment by muhammedfurkanyilmaz** [comment]
Hİ @avdg81,
Thanks for the instant response. I have an additional question related to this. What types of boundaries are suitable for dynamic analysis in geomechanics? or which boundaries have you used before in such analyses? 

Regards
Furkan

**Comment by loumalouomega** [comment]
@avdg81 if you have questions regarding the MPC you can ask me, it is a subject that I have quite fresh as recently I added the MPI support for it.

**Reply by avdg81** [reply]
Hi @loumalouomega,

Thank you for your kind offer. I just gave two pointers that may help @muhammedfurkanyilmaz to continue with his analyses. As indicated, we have no experience with the multipoint constraints in Kratos ourselves. So depending on whether he needs further assistance, we may or may not need your expertise in this area. We will notify you in case we do. Thanks.

Kind regards,
Anne
